pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -56,27 +56,30 @@ type ClientConfig interface {
 	Namespace() (string, bool, error)
 }
 
+type PersistAuthProviderConfigForUser func(user string) restclient.AuthProviderConfigPersister
+
 // DirectClientConfig is a ClientConfig interface that is backed by a clientcmdapi.Config, options overrides, and an optional fallbackReader for auth information
 type DirectClientConfig struct {
-	config         clientcmdapi.Config
-	contextName    string
-	overrides      *ConfigOverrides
-	fallbackReader io.Reader
+	config                    clientcmdapi.Config
+	contextName               string
+	overrides                 *ConfigOverrides
+	fallbackReader            io.Reader
+	persistAuthProviderConfig PersistAuthProviderConfigForUser
```

Comment:
> @deads2k you mean strongly-typed persisters? How would one do that in Go, given the lack of generics, templating, etc.

Yeah, that's what I meant, but without generics or templating I can see that the effort would be too great to be worth it.

My main fear is trying to put the burden of building the `persistAuthProviderConfigForUser` on the caller to a new config.  I don't think most callers will pass anything other than nil and that will produce problems later.